diff --git a/node_modules/pg/lib/index.js b/node_modules/pg/lib/index.js
index 7f02aba..bca7f8a 100644
--- a/node_modules/pg/lib/index.js
+++ b/node_modules/pg/lib/index.js
@@ -25,31 +25,4 @@ var PG = function (clientConstructor) {
   this.DatabaseError = DatabaseError
 }
 
-if (typeof process.env.NODE_PG_FORCE_NATIVE !== 'undefined') {
-  module.exports = new PG(require('./native'))
-} else {
-  module.exports = new PG(Client)
-
-  // lazy require native module...the native module may not have installed
-  Object.defineProperty(module.exports, 'native', {
-    configurable: true,
-    enumerable: false,
-    get() {
-      var native = null
-      try {
-        native = new PG(require('./native'))
-      } catch (err) {
-        if (err.code !== 'MODULE_NOT_FOUND') {
-          throw err
-        }
-      }
-
-      // overwrite module.exports.native so that getter is never called again
-      Object.defineProperty(module.exports, 'native', {
-        value: native,
-      })
-
-      return native
-    },
-  })
-}
+module.exports = new PG(Client)
